File: agent-training-management-api/Models/Entities/Class.cs
namespace AtmAPI.Models.Entities;

public class Class : EntityBase
{
	[MaxLength(ModelConstants.Commons.MaxLength)]
	public string Name { get; set; } = default!;

	[MaxLength(ModelConstants.Commons.DescriptionLength)]
	public string? Description { get; set; }

	[MaxLength(ModelConstants.Commons.MaxLength)]
	public string Placement { get; set; } = default!;
	public DateTime StartDate { get; set; }
	public DateTime? EndDate { get; set; }

	// [true] opening | [false] closed | [null] future
	public bool? Available { get; set; }
	public bool EnableAutomation { get; set; }
	public bool IsExternal { get; set; } = false;
	public int? MaxLearner { get; set; }
	public int SkillId { get; set; }

	public virtual Skill Skill { get; set; } = default!;
	public virtual ICollection<Schedule>? Schedules { get; set; }
	public virtual ICollection<SkillReport>? SkillReports { get; set; }
	public virtual ICollection<ClassLecturer>? ClassLecturers { get; set; }
}

File: agent-training-management-ui/src/types/entities/class.ts
import type { EntityBase } from '@/commons';

import type { LecturerResponse } from './lecturer';
import type { SkillResponse } from './skill';

export interface ClassUpsert {
	name?: string;
	placement?: string;
	start_date?: string | Date;
	end_date?: string | Date;
	available?: boolean;
	enable_automation?: boolean;
	max_learner?: number;
	skill_id?: number | string;
	description?: string;
}

export interface ClassResponse
	extends Omit<ClassUpsert, 'skill_id'>,
		EntityBase {
	skill: SkillResponse;
	lecturers: LecturerResponse[];
}
